gdal/frmts/netcdf/netcdflayer.cpp

```diff
@@ -2357,9 +2572,18 @@ OGRErr netCDFLayer::CreateField(OGRFieldDefn *poFieldDefn, int /* bApproxOK */)
                 size_t nDim = poFieldDefn->GetWidth() == 0
                                   ? m_nDefaultWidth
                                   : poFieldDefn->GetWidth();
+            if(!m_bLegacyCreateMode)
```

Comment:
see above remarks. could be merged in a single code path with a passthrough vcdf implementation